File: library/src/main/java/renetik/android/view/list/CSListRowView.java
package renetik.android.view.list;

import renetik.android.json.CSJsonData;
import renetik.android.viewbase.CSLayoutId;
import renetik.android.java.callback.CSRunWith;

/**
 * Created by renetik on 05/01/18.
 */

public class CSListRowView<DataType extends CSJsonData> extends CSRowView<CSListRow<DataType>> {
    private final CSRunWith<CSListRowView<DataType>> _onLoad;

    public CSListRowView(CSListController parent, CSLayoutId layout, CSRunWith<CSListRowView<DataType>> onLoad) {
        super(parent, layout);
        _onLoad = onLoad;
    }

    protected void onLoad(CSListRow<DataType> row) {
        super.onLoad(row);
        _onLoad.run(this);
    }

}

File: library/src/main/java/renetik/android/view/list/CSListRowView.kt
package renetik.android.view.list

import renetik.android.json.CSJsonData
import renetik.android.viewbase.CSLayoutId

class CSListRowView<DataType : CSJsonData>(
        parent: CSListController<CSListRow<DataType>, *>, layout: CSLayoutId,
        private val onLoad: (CSListRowView<DataType>) -> Unit)
    : CSRowView<CSListRow<DataType>>(parent, layout) {

    override fun onLoad(row: CSListRow<DataType>) = onLoad.invoke(this)
}
